lib/iris/plot.py

```diff
@@ -102,10 +102,20 @@ def get_span(coord):
             raise ValueError(msg.format(coord.name()))
         if mode == iris.coords.BOUND_MODE and len(span) not in [1, 2]:
             raise ValueError('The coordinate {!r} has {} dimensions.'
-                             'Cell-based plotting is only supporting for'
+                             'Cell-based plotting is only supported for'
                              'coordinates with one or two dimensions.'
                              .format(coord.name()), len(span))
 
+    # Check the coords have the same number of dimensions.
+    if ndims == 2:
+        if not (len(spans[0]) == len(spans[1])):
+            raise ValueError('The coordinate {!r} has {} dimensions and the '
+                             'coordinate {!r} has {} dimensions. Cell-based '
+                             'plotting is only supported for coordinates with '
```

Comment:
I'd like to discuss this with you, because I am a little unsure of what cell-based plotting means. Also whether it is clear to users what cell-based plotting is.  And whether it matters, I suppose.